src/Controls/src/Core/TemplatedItemsList.cs

```diff
@@ -534,8 +534,14 @@ public DataTemplate SelectDataTemplate(object item)
 
 		public TItem ActivateContent(int index, object item)
 		{
-			TItem content = ItemTemplate != null ? (TItem)ItemTemplate.CreateContent(item, _itemsView) : _itemsView.CreateDefault(item);
-
+			if (ItemTemplate?.CreateContent(item, _itemsView) is TItem content)
+			{
+				// content is valid
+			}
+			else
+			{
+				content = _itemsView.CreateDefault(item);
+			}
```

Comment:
Why don't you negate the if so that it is more clear, looks strange to have an empty {}
```suggestion
			if (!(ItemTemplate?.CreateContent(item, _itemsView) is TItem content))
			{
				// content is not valid, use the defaul
				content = _itemsView.CreateDefault(item);
			}
```

Reply:
So I think I can use the new C# feature `is not`:
```code
if (ItemTemplate?.CreateContent(item, _itemsView) is not TItem content)
```
Let me just test it locally.